cadc-web-util/src/main/java/ca/nrc/cadc/accesscontrol/AccessControlClient.java

```diff
@@ -95,22 +104,106 @@ public void resetPassword(final char[] newPassword, final char[] token) {
         final Map<String, String> headers = new HashMap<>();
         headers.put(CADC_TOKEN_HEADER_KEY, new String(token));
 
-        final int statusCode = post(lookupPasswordResetURL(), payload, headers, out);
-        switch (statusCode) {
-            case 200: {
-                break;
-            }
+        Subject subject = AuthenticationUtil.getAnonSubject();
 
-            case 403:
-            case 401: {
-                throw new AccessControlException("Login denied");
-            }
+        try {
 
-            default: {
-                throw new IllegalArgumentException(
-                    String.format("Unable to reset password.\nServer error code: %d.", statusCode));
+            Subject.doAs(subject, new PrivilegedExceptionAction<Void>() {
+
+                @Override
+                public Void run() throws Exception {
+                    final HttpPost thePost = postNoRedirect(lookupPasswordResetURL(), payload, headers);
+                    int statusCode = thePost.getResponseCode();
+                    StringBuilder logStr = new StringBuilder();
+                    logStr.append("Unable to reset password");
+                    StringBuilder throwStr = new StringBuilder();
+                    throwStr.append("Unable to reset password");
+
+                    // Gets added after specific message
+                    StringBuilder reasonPart = new StringBuilder();
+                    if (statusCode != 200) {
+                        reasonPart.append(" - ");
+                        reasonPart.append(thePost.getResponseCode());
+                        reasonPart.append(": ");
+                        reasonPart.append(thePost.getThrowable().toString());
```

Comment:
I think the interesting message is in the response body which can be retrieved from the input stream...

Reply:
Or you could use the deprecated method getResponseBody()